Dockerfile

```diff
@@ -7,45 +7,44 @@ RUN pip install poetry
 # copy all except in dockerignore
 COPY . .
 # install prerequisits
-RUN poetry install --no-root --no-dev
-#RUN pip install -r requirements.txt
-# use port 5000 from container
+RUN poetry config virtualenvs.create false --local && poetry install
+# use port from container
 EXPOSE 5000
 
-
-
-# prod image
-FROM base as production
-# install gunicorn
-RUN pip install gunicorn
-# Cmd /entrypoint
-CMD ["gunicorn"  , "-b", "0.0.0.0:5000", "todo_app.app:create_app()"]
-
-# dev image
+# development - image
 FROM base as development
 # install flask
 #RUN pip install flask
-
 CMD [ "poetry", "run", "flask", "run", "--host=0.0.0.0", "--port=5000", "--debugger"] 
 
-# testing stage
+# testing Image
 FROM base as test
 RUN pip install pytest
 #RUN poetry Install
-RUN poetry add pytest --dev
+RUN poetry add pytest --group dev
 #WORKDIR /tests
 ENV GECKODRIVER_VER v0.31.0
 # Install the long-term support version of Firefox (and curl if you don't have it already)
 RUN apt-get update && apt-get install -y firefox-esr curl
-  
 # Download geckodriver and put it in the usr/bin folder
 RUN curl -sSLO https://github.com/mozilla/geckodriver/releases/download/${GECKODRIVER_VER}/geckodriver-${GECKODRIVER_VER}-linux64.tar.gz \
    && tar zxf geckodriver-*.tar.gz \
    && mv geckodriver /usr/bin/ \
    && rm geckodriver-*.tar.gz
+# install Selenium
 RUN  pip install selenium   
 ENTRYPOINT ["poetry", "run", "pytest"]
 
+# production image
+FROM base as production
+# install gunicorn
+RUN pip install gunicorn
+# Cmd /entrypoint
+#CMD poetry run gunicorn "todo_app.app:create_app()" --bind 0.0.0.0:$PORT
+CMD gunicorn -b 0.0.0.0:$PORT "todo_app.app:create_app()"
```

Comment:
Note that this change to use $PORT (as required by Heroku) breaks your Readme instructions for how to run the production container since we don't set a value. A nice solution can be to set a default value as before that Heroku can override, for example:
```
ENV PORT=5000
CMD gunicorn -b 0.0.0.0:$PORT "todo_app.app:create_app()"
```

Reply:
updated